ocs_ci/deployment/disconnected.py

```
@@ -256,31 +309,59 @@ def mirror_index_image_via_oc_mirror(index_image, packages, idms=None):
+        if os.path.exists(registries_template):
+            try:
+                with open(registries_template, "r") as f:
+                    registries_content = f.read()
+
+                # Backup and update registries.conf
+                registries_conf_path = "/etc/containers/registries.conf"
+                exec_cmd(
+                    f"sudo cp {registries_conf_path} {registries_conf_path}.backup"
+                )
+
+                temp_file = tempfile.NamedTemporaryFile(
```

Ruff (0.15.13)

[warning] Use a context manager for opening files

(SIM115)